In [1]:
%load_ext autoreload
%autoreload

import sys

sys.path.append('../../utils')
from utils import PCA_reduction

### PCA

In [2]:
PCA_reduction('../../../dumps/various_sizes/8K.csv','tree')

  Variance    Training acc    Test acc    Components    Time (s)
----------  --------------  ----------  ------------  ----------
      1           0.986137    0.977946           119    0.492519
      0.99        0.985507    0.980466           103    0.429799
      0.95        0.985665    0.979836            81    0.353857
      0.9         0.986452    0.981096            64    0.287772
      0.85        0.986137    0.978576            51    0.243596


In this case the accuracies are really similar, even slightly better, but we managed to reduced the number of features from 119 to 47, so more than 60%. The time as also be improved by half !